src/baseline/expr.rs

```diff
@@ -778,6 +778,19 @@ where
         if let Some(intrinsic) = self.intrinsic(e.id) {
             self.emit_intrinsic_bin(&e.lhs, &e.rhs, dest, intrinsic, Some(e.op));
         } else if e.op == BinOp::Cmp(CmpOp::Is) || e.op == BinOp::Cmp(CmpOp::IsNot) {
+            let mode = self.ty(e.lhs.id()).mode();
+            if mode.is_float() {
+                let intrinsic = match mode {
+                    MachineMode::Float32 => Intrinsic::FloatAsInt,
+                    MachineMode::Float64 => Intrinsic::DoubleAsLong,
+                    _ => unreachable!()
+
+                };
+                // expects &ExprCallType, e is Box<Expr>
+                //self.emit_intrinsic_float_as_int(e.lhs, REG_TMP1, intrinsic);
+                //self.emit_intrinsic_float_as_int(e.rhs, REG_TMP2, intrinsic);
```

Comment:
This is the place I'm stuck... help appreciated.

If I get floating-point values here, I want to convert them to integer values (with `movd`, as implemented earlier) and do a normal integer comparison on them.

Reply:
Don't go through an intrinsic here, you can directly MacroAssembler methods, I think it should look something like this pseudocode:

```
self.emit_expr(&e.lhs, FREG_RESULT.into()); // result is a float
mov_d2i(REG_RESULT, FREG_RESULT);
// save REG_RESULT on stack

self.emit_expr(&e.rhs, FREG_RESULT.into()); // result is a float
mov_d2i(REG_RESULT, FREG_RESULT);
// now compare REG_RESULT with value on stack
```

Reply:
Oh, forgot to mention: The non-float code path already stores a register on the stack and then compares both values. You can lookup the necessary code for these two comments above from there.

Reply:
Thanks, I'll check it out!

Reply:
Thanks for the help! :-)